src/Raven.Server/Documents/DocumentPutAction.cs

```diff
@@ -54,11 +56,44 @@ public DocumentPutAction(DocumentsStorage documentsStorage, DocumentDatabase doc
             }
         }
 
+        private struct CompareClusterTransactionId
+        {
+            private readonly ServerStore _serverStore;
+            private readonly DocumentPutAction _parent;
+
+            public CompareClusterTransactionId(DocumentPutAction parent)
+            {
+                _serverStore = parent._documentDatabase.ServerStore;
+                _parent = parent;
+            }
+
+            public void ValidateAtomicGuard(string id, NonPersistentDocumentFlags nonPersistentDocumentFlags, string changeVector)
+            {
+                if (nonPersistentDocumentFlags != NonPersistentDocumentFlags.None) // replication or engine running an operation, we can skip checking it 
```

Comment:
`NonPersistentDocumentFlags` has _very_ different options. We might add some additional flags in the future. 

Is it okay to say that we don't validate this guard if _anything_ is set?

Reply:
I think so.
My worry is that we may break other features because of this.
Resolving conflicts, coming from replication, etc.

The error we raise here is mostly meant to help detect race conditions, and many of those issues aren't really possible to seamlessly resolve.
I would rather have it skip in most cases, since if this happened a few seconds later, it will go through with no issue, mostly likely.

Reply:
@karmeli87 your opinion about that?

Reply:
Not sure to be honest... the state here might be very complex and different, so I understand the worry of breaking some features.

For fun, I would maybe try to change the default of all tx to be cluster-wide. 
Especially for tests that are testing conflicts... usually we do that with external replication, which shouldn't break anything here, I think.
I do expect some tests to fail... but most of them should pass, I think.